kube_controller_manager/tests/test_sli_metrics.py

```diff
@@ -0,0 +1,168 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import os
+
+import mock
+import pytest
+import requests_mock
+
+from datadog_checks.kube_controller_manager import KubeControllerManagerCheck
+
+from .common import HERE
+
+# Constants
+CHECK_NAME = 'kube_controller_manager'
+
+
+@pytest.fixture()
+def mock_metrics():
+    f_name = os.path.join(HERE, 'fixtures', 'metrics_slis_1.27.3.txt')
+    with open(f_name, 'r') as f:
+        text_data = f.read()
+    with mock.patch(
+        'requests.get',
+        return_value=mock.MagicMock(
+            status_code=200, iter_lines=lambda **kwargs: text_data.split("\n"), headers={'Content-Type': "text/plain"}
+        ),
+    ):
+        yield
+
+
+def test_check_metrics_slis(aggregator, mock_metrics, mock_request, instance):
+    mock_request.get('http://localhost:10257/metrics/slis', status_code=200)
+    c = KubeControllerManagerCheck(CHECK_NAME, {}, [instance])
+    c.check(instance)
+
+    def assert_metric(name, **kwargs):
+        # Wrapper to keep assertions < 120 chars
+        aggregator.assert_metric("{}.{}".format(CHECK_NAME, name), **kwargs)
+
+    assert_metric(
+        'slis.kubernetes_healthcheck', value=1, metric_type=aggregator.GAUGE, tags=['name:attachdetach', 'type:healthz']
```

Comment:
> For visibility (from slack discussion): the [kubernetes documentation](https://kubernetes.io/docs/reference/instrumentation/slis/#sli-metrics) `readyz` is a potential value for `type`. However, as these metrics are health-specific, only samples with `type=healthz` should be kept. In the future, a new metric can be added for samples with different values for `type`.

Follow up question: should the `type` tag be removed from this metric since only `type=healthz` would be kept?